### ISSUE TITLE
Windows Shell Baloon Messages Stick on Screen

### ISSUE THREAD

**Issue by jebeld17** [canton7/SyncTrayzor#398]
Often times when I hover over the Synctrayzor icon in the Windows System Tray, the balloon message stating what program it is that I'm hovering over will pop up in the upper left-hand corner of the screen and stick there, regardless if I move my mouse, open Synctrayzor, or interact with the tray icon in any way, shape or form.

The only way I found to fix this is to exit out of Synctrayzor completely and re-open the program.
Has anyone else run into this?

Windows 10 Home x64, v1703
![synctrayzor_balloon_stuck](https://user-images.githubusercontent.com/9492083/29676315-b8518642-88c5-11e7-9278-74007fa62112.PNG)

**Comment by canton7** [comment]
I'm afraid I've never seen it. I'll leave this open in case anyone else has come across it.

**Comment by jebeld17** [comment]
I've run across this on all Windows 10 x64 machines running Version 1703, whether it's on Home or Pro.

**Comment by canton7** [comment]
Weird, I struggle to imagine how this is anything which I can affect, since that tooltip is provided solely by Windows.

**Comment by jebeld17** [comment]
Maybe we can either change the libraries *or* set a fast refresh cycle for the libraries? (Basically have the tray icon itself shut down for a brief half-second every "x" amount of time)

**Comment by canton7** [comment]
No, this is the only notify icon library for WPF. There's no such thing as a refresh cycle for a notify icon.

**Comment by jebeld17** [comment]
Worth a shot to ask, at least :(

**Comment by msrv** [comment]
Same problem here! Windows 10 & Surface Pro. Tooltip stays in top left corner...which external library is used? maybe we can open the issue there...

**Comment by jebeld17** [comment]
Thank you @msrv

**Comment by canton7** [comment]
Project page is here: http://www.hardcodet.net/wpf-notifyicon . There is no issue tracker.

**Comment by jebeld17** [comment]
I sent the developer an email.

**Comment by Hard-Wired** [comment]
Confirmed on my Windows 10 1709 16299.248 / Synctrayzor: v1.1.18 / Syncthing: v0.14.44

I can sometimes get it to repeat if I futz with the tray icon or hover over the tray icon while locking the desktop.

**Comment by cosmicdan** [comment]
Confirmed here too. I only just started noticing this today. On my home PC and Notebook it happens to both, the Notebook being only a day old and a fairly-clean Windows 10 home install, as well as the portable version on my USB that I used just now on public Uni PC's.

**Comment by Hard-Wired** [comment]
I've found that if I hover the mouse over a different icon in the task tray, the stuck SyncTrayzor message disappears once the other icon's message is activated. Not a fix, but at least it gets rid of the message.

**Comment by vvvlad** [comment]
I can confirm that this is an issue with the hardcodet wpf library, as it have been observed with other applications using this library.
It was not updated since 2016, and the developer seems not to have been very active since then.
Very strange this is the only option available. :(

**Comment by ambrosegarnet** [comment]
I also have this issue intermittently. New to github, don't know what else to say??

**Comment by jebeld17** [comment]
Maybe we should invest in a fork of hardcodet wpf? Can't hurt to help out a few other projects if any of us get free time to contribute. :)

**Comment by DieKatzchen** [comment]
There's half a dozen pull requests that haven't been addressed. I'm not sure what the protocol is for taking over an open source project after the author has dropped off the earth is.

**Comment by MyFault** [comment]
> 
> 
> I've found that if I hover the mouse over a different icon in the task tray, the stuck SyncTrayzor message disappears once the other icon's message is activated. Not a fix, but at least it gets rid of the message.

Thank you! I was struggling with this issue, and absent an actual fix (which seems unlikely due to the 3rd party library), this workaround is extremely helpful!

**Comment by canton7** [comment]
Just an observation that wpf.notifyicon is now on github: https://github.com/hardcodet/wpf-notifyicon

**Comment by jebeld17** [comment]
👍

**Comment by TheCoinAxe** [comment]
I'm having the same issue, is this fixed ?

**Comment by canton7** [comment]
Unless you can find an indication that an update to the wpf.notifyicon library has fixed it, nope!

**Comment by bitfiddlers** [comment]
FYI: Just observed same behaviour from NordVPN Tray app on my Surface Pro 5

**Comment by hifialan** [comment]
Easiest way to get rid of it for me is to right click the Syncthing tray icon and select "Restart Syncthing".

**Comment by 50705** [comment]
I had the same issue with NordVPN and it has nothing to do with the SyncTrayzor. 
If you run this on a system with a dedicated GPU consider the following actions:
1. Add the App and service executables to the "High Performance" mode of the  Windows Settings>Display>Graphics Settings>Graphics Performance Preference
2. Properties the App and service executables must have the 
 a. Properties>Compatibility>Disable Fullscreen Optimizations
 b. Change High DPI settings [v] Program DPI
 c. Override High DPI scaling behavior [v] 
 d. Scaling performed by application
3. Stop and start the service and the problem must have disappeared.

**Comment by Nova38** [comment]
I agree that that might be a good solution for the time being

**Comment by xgdgsc** [comment]
Has anyone tried using https://github.com/hardcodet/wpf-notifyicon latest code and test this issue? There' s no related issue open on the new repo. If still occurs, we should open one issue there.

**Comment by acomagu** [comment]
The sample apps in the repository of wpf-notifyicon don't seem to reproduce this issue.

![image](https://user-images.githubusercontent.com/1362750/232227610-e918edd1-95bf-4381-84c4-b4487175d9de.png)

However, I noticed the library seems to draw tooltips in its own way, without using the Windows native tooltip UI. I think that is causing problems and is a complexity that SyncTrazor does not need.

Native's tooltip UI looks more modern in Windows 11.

![image](https://user-images.githubusercontent.com/1362750/232228473-03b80e91-8660-4604-b8bd-5e07347e92f2.png)

I think it would be nice to find a way to use Native's tooltip UI.

(Technical info: To show native tooltip, add `NIF_SHOWTIP` flag to `Shell_NotifyIcon` calling. [Document](https://learn.microsoft.com/windows/win32/api/shellapi/ns-shellapi-notifyicondataa#nif_showtip-0x00000080))

**Comment by TheRealKamisama** [comment]
Same issue on Windows 11 22H2 2621.1555

**Comment by cartermnelson** [comment]
Same issue here as well on Windows 11 22H2 22621.1848 on multiple systems

**Comment by fhumayun** [comment]
Can confirm this is reproducible, and still annoying 😒, 
OS Name **Microsoft Windows 11 Pro**
Version **10.0.22621 Build 22621**

**Comment by beyond9thousand** [comment]
has anyone figured this out? its pretty annoying

**Comment by Hollo1001** [comment]
Many apps I use have this issue, its super annoying.
I will just stop using them, when I don't 100% need them.

**Comment by yatimerkhanov** [comment]
I would greatly appreciate it if this could be addressed. I encounter this balloon message daily.

**Comment by simon-nicalis** [comment]
There is this PR on the wpf-notifyicon repo which might solve the issue: https://github.com/hardcodet/wpf-notifyicon/pull/96

**Comment by BalduinMeyer** [comment]
I got rid of it by changing compatibility settings of "C:\Program Files\SyncTrayzor\SyncTrayzor.exe": set DPI scaling to "System". Now the icon is a bit blurry but that is a far minor problem.

**Comment by Abendstern51** [comment]
I've come to the following. Create a second Desktop and open SyncTrayzor as a window on it. The annoying icon will disappear and SyncTrayzor won't bother you on the first main Desktop. I do this every time I lauch my computer.

**Comment by mitchh** [comment]
> I got rid of it by changing compatibility settings of "C:\Program Files\SyncTrayzor\SyncTrayzor.exe": set DPI scaling to "System". Now the icon is a bit blurry but that is a far minor problem.

Con confirm, this fixed it for me. Thanks @BalduinMeyer
